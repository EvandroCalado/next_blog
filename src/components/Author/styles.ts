import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 3rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 15rem;
    border: 3px solid ${theme.colors.secondary};
    border-radius: 50%;
  `}
`;

export const Container = styled.div`
`;

export const Title = styled.h2`
${({ theme }) => css`
font-size: ${theme.font.sizes.xlarge};
`}
`;

export const Paragraph = styled.p`
`;
