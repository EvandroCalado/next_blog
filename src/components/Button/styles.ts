import styled, { css } from 'styled-components';
import { ButtonProps } from './';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background-color: ${theme.colors[color]};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    border: none;
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    margin: ${theme.spacings.large} 0;
    border-radius: 0.5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 300ms ease-in-out;
    cursor: pointer;

    &:hover {
      filter: brightness(80%);
    }

    &:disabled {
      background-color: ${theme.colors.mediumGray};
      color: ${theme.colors.darkGray};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 3rem;
      margin-left: 1rem;
    }
  `}
`;
