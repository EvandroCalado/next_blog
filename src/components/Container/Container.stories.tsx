import Container, { ContainerProps } from './Container';
import { Meta, StoryFn } from '@storybook/react';

export default {
  title: 'Container',
  component: Container,
  args: {
    children: 'Contaier',
  },
} as Meta;

export const Template: StoryFn<ContainerProps> = (args) => (
  <div>
    <Container {...args} />
  </div>
);
