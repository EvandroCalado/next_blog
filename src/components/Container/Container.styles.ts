import styled, { css } from 'styled-components';

export const Wrapper = styled.main`
  ${({ theme }) => css`
    max-width: 96rem;
    margin: 0 auto;
    padding: ${theme.spacings.medium};
    font-size: ${theme.font.sizes.small};
  `}
`;
