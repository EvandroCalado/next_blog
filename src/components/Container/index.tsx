import * as Styled from './styles';

export type ContainerProps = {
  children: React.ReactNode;
};

const Container = ({ children }: ContainerProps) => {
  return <Styled.Wrapper>{children}</Styled.Wrapper>;
};

export default Container;
