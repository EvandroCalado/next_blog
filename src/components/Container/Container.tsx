import * as Styled from './Container.styles';

export type ContainerProps = {
  children: React.ReactNode;
};

const Container = ({ children }: ContainerProps) => {
  return <Styled.Wrapper>{children}</Styled.Wrapper>;
};

export default Container;
