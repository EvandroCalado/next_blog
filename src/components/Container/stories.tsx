import Container, { ContainerProps } from '.';
import { Meta, StoryFn } from '@storybook/react';

export default {
  title: 'Container',
  component: Container,
  args: {
    children: 'Contaier',
  },
} as Meta;

export const Template: StoryFn<ContainerProps> = (args) => (
  <div>
    <Container {...args} />
  </div>
);
