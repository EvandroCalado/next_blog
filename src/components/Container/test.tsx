import Container from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

describe('<Container />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Container>{'Container'}</Container>);
    expect(screen.getByText('Container')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
