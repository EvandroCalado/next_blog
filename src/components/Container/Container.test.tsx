import Container from './Container';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

describe('<Container />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Container>{'Container'}</Container>);
    expect(screen.getByText('Container')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
