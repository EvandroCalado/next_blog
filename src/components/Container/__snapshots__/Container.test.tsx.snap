// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Container /> > should render 1`] = `
.c0 {
  max-width: 96rem;
  margin: 0 auto;
  padding: 2.4rem;
  font-size: 1.6rem;
}

<div>
  <main
    class="c0"
  >
    Container
  </main>
</div>
`;
